pkgs/applications/editors/o/default.nix

```diff
@@ -0,0 +1,39 @@
+{ lib, stdenv, buildGoModule, fetchFromGitHub, installShellFiles, makeWrapper, pkg-config
+, withGui ? stdenv.isLinux, vte # vte is broken on darwin
+}:
+
+buildGoModule rec {
+  pname = "o";
+  version = "2.55.1";
+
+  src = fetchFromGitHub {
+    owner = "xyproto";
+    repo = "o";
+    rev = "v${version}";
+    hash = "sha256-owueLd6kR/bDFxKI9QOUgriH63XRsEEpIFfp5aRTSbI=";
+  };
+
+  vendorSha256 = null;
+
+  nativeBuildInputs = [ installShellFiles makeWrapper pkg-config ];
+
+  buildInputs = lib.optional withGui vte;
+
+  preBuild = "cd v2";
+
+  postInstall = ''
+    cd ..
+    installManPage o.1
+  '' + lib.optionalString withGui ''
+    make install-gui PREFIX=$out
+    wrapProgram $out/bin/ko --prefix PATH : $out/bin
```

Comment:
Why do we do this? If o is run from PATH this is already in PATH.

Reply:
Yes, it's for cases `o` is not in PATH, let's say `./result/ko`.

Reply:
Sure, we can, I just tried to avoid any source code patching to make maintanence easy in future. What do you think? Is it really better to hardcode path?

Reply:
> Also https://github.com/xyproto/o/blob/main/ko/main.cpp#L591-L594 will not work

Fixed